### ISSUE TITLE
Feature Idea: Override Empty Values

### ISSUE THREAD

**Issue by michael-e** [brendo/default_event_values#6]
If values are set in the POST array but empty, the extension will not override them (which is the right behaviour, of course). It would be cool to have an option for this, in addition to the _Value will override any value posted from the frontend_ option. It might be called _Value will override empty value posted from the frontend_ or similar.

**Comment by brendo** [comment]
Yeah, that'd be pretty neat, so you could always provide a default even if the frontend tried to clear the value. Could be useful for security too.

**Comment by DavidOliver** [comment]
> If values are set in the POST array but empty, the extension will not override them (which is the right behaviour, of course).

Maybe a silly question, but why is this the right behaviour?

Currently, a malicious user can remove an entry's field's data if they know/guess the field name and include an empty value with the POST?

**Comment by michael-e** [comment]
Yes, basically it is the right behaviour. The extension is called "Default Event Values", so it is supposed to provide default values that are used if nothing is posted.

The option "Value will override any value posted from the frontend" can be used to make the default value override the posted value.

My suggestion means something in between: Override empty values, but don't override non-empty values. There are rare use cases for this (but I don't remember mine, unfortunately).

**Comment by DavidOliver** [comment]
> My suggestion means something in between

I see. So the current 'Value will override any value posted from the frontend' option can already be used to override empty values? Thanks.

**Comment by michael-e** [comment]
Yes, as it says: _override any value_. :-)

**Comment by DavidOliver** [comment]
Good good! The first sentence of this issue threw me.

**Comment by michael-e** [comment]
:-) (Speak: Yes, it was misleading.)
